libalpm/scripts/mkinitcpio-install

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+
+args=()
+all=0
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
+        all=1
+        continue
+    fi
+
+    pkgbase=$(<"${line%/vmlinuz}/pkgbase")
+    [[ -n ${pkgbase} ]]
+
+    preset="/etc/mkinitcpio.d/${pkgbase}.preset"
+    if [[ ! -e $preset ]]; then
+        sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset \
+            | install -Dm644 /dev/stdin "$preset"
```

Comment:
If some nonconformant kernel package does not do what it is expected to do and fails to install a mkinitcpio preset, we can just skip it. Please don't go install configuration files behind my back that aren't tracked by pacman -- it would cause file conflicts when the kernel package gets updated to include the mkinitcpio preset due to my complaining to the maintainer. :p

If you absolutely must support nonconforming packages, install the hook to /tmp or /run, and call mkinitcpio with the full absolute filename.

Reply:
This goes along with changes to the kernels which remove all files outside `/usr/lib/modules/$kernver/`.

Reply:
Well, we discussed this at length, and all the official kernels at least, will not have anything outside /usr/lib/modules. It will be either mkinitcpio's or dracut's job to handle the actual kernel installation and initramfs generation. I'm still working on dracut hooks and we are discussing even more changes for that. I guess I can add a check, for example, that if a package has vmlinuz (most will), but doesn't have the pkgbase file, we bail and let that kernel do their own thing. Which, might as well include their own mkinitcpio hook and preset.

Reply:
Doing it this way decouples our kernels from the way they are installed into `/boot/`. This will allow us to implement better ways in `dracut` or other packages.

Reply:
So we're making tons of sweeping changes all at once, including moving around the vmlinuz file in ways that mess up peoples' expectations and don't even fit the purpose of lib/modules anyway, in order to implement mysterious "better ways" that are probably the inferior `kernel-install` which isn't even needed for dracut?

Not a fan at all, but either way I think we should make incremental changes, focus on making the hooks better, and then re-envision how we should install (or fail to install) kernels for all of arch.

Reply:
I've added the check for non pkgbase kernels with 9c1e0f3dd01151b79e9b557ea8510605616c1369.